vernac/prettyp.ml

```diff
@@ -208,6 +208,54 @@ let print_polymorphism env ref =
          ++ if !Detyping.print_universes then h (pr_template_variables template_variables) else mt()
        else str "not universe polymorphic") ]
 
+let print_squash env ref udecl = match ref with
+  | GlobRef.IndRef ind ->
+    let _, mip = Inductive.lookup_mind_specif env ind in
+    begin match mip.mind_squashed with
+    | None -> []
+    | Some squash ->
+      let univs = Environ.universes_of_global env ref in
+      let bl = Printer.universe_binders_with_opt_names univs udecl in
+      let sigma = Evd.from_ctx (UState.of_names bl) in
+      let inst = if fst @@ UVars.AbstractContext.size univs = 0 then mt()
+        else Printer.pr_universe_instance sigma (UVars.make_abstract_instance univs)
+      in
+      let inds = match mip.mind_arity with
+        | TemplateArity _ -> assert false
+        | RegularArity a -> a.mind_sort
+      in
+      let target = match inds with
+          | SProp -> str "SProp"
+          | Prop -> str "SProp or Prop"
+          | Set -> str "SProp, Prop or Set"
+          | Type _ -> str "not in a variable sort quality"
+          | QSort (q,_) -> str "in sort quality " ++ Termops.pr_evd_qvar sigma q
+      in
+      let unless = match squash with
+        | AlwaysSquashed -> str "."
+        | SometimesSquashed qs ->
+          let target = match inds with
+            | SProp | Prop | Set -> target
+            | Type _ -> str "instantiated to constant qualities"
+            | QSort (q,_) -> str "smaller than the instantiation of " ++ Termops.pr_evd_qvar sigma q
```

Comment:
"smaller" is confusing here. I think we should use another word, like "eliminable" or something like it.

Reply:
That's not defined anywhere though I think